src/Compilers/CSharp/Portable/Emitter/Model/PEModuleBuilder.cs

```diff
@@ -208,6 +209,94 @@ internal sealed override IEnumerable<Cci.INestedTypeDefinition> GetSynthesizedNe
             return null;
         }
 
+        public sealed override List<(Cci.IDefinition definition, List<Cci.DebugSourceDocument> document)> GetTypeDocument(EmitContext context)
+        {
+            var result = new List<(Cci.IDefinition definition, List<Cci.DebugSourceDocument> document)>();
+
+            var namespacesAndTypesToProcess = new Stack<NamespaceOrTypeSymbol>();
+            namespacesAndTypesToProcess.Push(SourceModule.GlobalNamespace);
+            while (namespacesAndTypesToProcess.Count > 0)
+            {
+                NamespaceOrTypeSymbol symbol = namespacesAndTypesToProcess.Pop();
+                var locations = symbol.Locations.ToArray();
+                switch (symbol.Kind)
+                {
+                    case SymbolKind.Namespace:
+                        if (!locations.IsEmpty())
+                        {
+                            foreach (var member in symbol.GetMembers())
+                            {
+                                switch (member.Kind)
+                                {
+                                    case SymbolKind.Namespace:
+                                    case SymbolKind.NamedType:
+                                        namespacesAndTypesToProcess.Push((NamespaceOrTypeSymbol)member);
+                                        break;
+                                    default:
+                                        throw ExceptionUtilities.UnexpectedValue(member.Kind);
+                                }
+                            }
+                        }
+                        break;
+                    case SymbolKind.NamedType:
+                        if (!locations.IsEmpty())
+                        {
+                            var docList = new HashSet<Cci.DebugSourceDocument>();
+                            if (!TypeHasIL(symbol, context, docList))
+                            {
+                                AddDefinitionAndDocument(result, locations, (Cci.IDefinition)symbol.GetCciAdapter(), docList);
+                            }
+                        }
+                        break;
+                    default:
+                        throw ExceptionUtilities.UnexpectedValue(symbol.Kind);
+                }
+            }
+            return result;
+        }
+
+        // Method checks if all methods contained in the type have IL, methods with IL documents will be added to a document list.
+        // The method will return true if all methods in the type have IL, otherwise will return false.
```

Comment:
>The method will return true if all methods in the type have IL, otherwise will return false. [](http://example.com/codeflow?start=11&length=92)

The comment on the method does not match the implementation.